### PR TITLE
Add better diff output for string comparison.

### PR DESCRIPTION
**Pull request by Ferada**
Will always output colours.  That's a point for improvement and should likely be configurable and/or determined by the output/terminal capabilities.  In general the library also supports custom output (e.g.
a HTML one is available).

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -219,7 +229,7 @@ func (hc *HttpCall) ResponseHeaderEquals(key, value string) Step {
 		if err := hc.EnsureResponse(); err != nil {
 			return err
 		} else if header := hc.Response.Header.Get(key); header != value {
-			return fmt.Errorf("Header '%s': Expected '%s'; found '%s'.", key, value, header)
+			return fmt.Errorf("Header: '%s': Differed: '%s'.", key, diff(value, header))
```
Comment (msackman) [inline http.go]: "Differed" seems an unusual word to me - just "Diff"?

File: http.go [hunk 2]
```diff
@@ -249,8 +259,8 @@ func (hc *HttpCall) ResponseBodyEquals(value string) Step {
 	return NewNamedStep("ResponseBodyEquals", func() error {
 		if err := hc.ReceiveBody(); err != nil {
 			return err
-		} else if string(hc.ResponseBody) != value {
-			return fmt.Errorf("Body: Expected '%s'; found '%s'.", value, string(hc.ResponseBody))
+		} else if got := string(hc.ResponseBody); got != value {
+			return fmt.Errorf("Body: Differed: '%s'.", diff(value, got))
```
Comment (msackman) [inline http.go]: Same here. Also, s/got/bodyStr/ ?

**Review by msackman — CHANGES_REQUESTED** [review]
Couple of tiny suggestions. In general fine. Slightly vexing that we're trying to come up with structured in errors in general. Suggests a design wart to me - maybe the 'error' type just isn't the right one. But let's leave that for another day.

**Comment by Ferada** [comment]
It's not really about errors though, but any kind of potentially less structured output, or where you want to see the exact difference before parsing, especially since the `encoding/*` libraries are so lenient on parsing. I'll update rest.

Edit: Oh right I misunderstood you, yes, since it's more report generation I'd otherwise just print to a stream really, or if we really wanted different output formats perhaps a structured object which could be formatted according to the kind of output, plain text, HTML etc. But pretty overkill at the moment :)

**Review by msackman — APPROVED** [review]

**Pull request merged**